multimodal/src/autogluon/multimodal/learners/base.py

```diff
@@ -0,0 +1,2473 @@
+                UserWarning,
+            )
+        return mixup_active, mixup_func
+
+    def get_loss_func_per_run(self, config, mixup_active=None):
+        loss_func = get_loss_func(
+            problem_type=self._problem_type,
+            mixup_active=mixup_active,
+            loss_func_name=OmegaConf.select(config, "optimization.loss_function"),
+            config=config.optimization,
+        )
+        return loss_func
+
+    def get_model_postprocess_fn_per_run(self, loss_func):
+        model_postprocess_fn = get_model_postprocess_fn(
+            problem_type=self._problem_type,
+            loss_func=loss_func,
+        )
+        return model_postprocess_fn
+
+    def get_datamodule_per_run(
+        self,
+        df_preprocessor,
+        data_processors,
+        per_gpu_batch_size,
+        num_workers,
+        predict_data=None,
+        is_train=True,
+    ):
+        if self._teacher_learner is not None:
+            df_preprocessor = [df_preprocessor, self._teacher_learner._df_preprocessor]
+            data_processors = [data_processors, self._teacher_learner._data_processors]
+        datamodule_kwargs = dict(
+            df_preprocessor=df_preprocessor,
+            data_processors=data_processors,
+            per_gpu_batch_size=per_gpu_batch_size,
+            num_workers=num_workers,
+        )
+        if is_train:
+            datamodule_kwargs.update(dict(train_data=self._train_data, validate_data=self._tuning_data))
+        else:
+            datamodule_kwargs.update(dict(predict_data=predict_data))
+
+        datamodule = BaseDataModule(**datamodule_kwargs)
+        return datamodule
+
+    def get_optimization_kwargs_per_run(self, config, validation_metric, custom_metric_func, loss_func, mixup_func):
+        return dict(
+            optim_type=config.optimization.optim_type,
+            lr_choice=config.optimization.lr_choice,
+            lr_schedule=config.optimization.lr_schedule,
+            lr=config.optimization.learning_rate,
+            lr_decay=config.optimization.lr_decay,
+            end_lr=config.optimization.end_lr,
+            lr_mult=config.optimization.lr_mult,
+            weight_decay=config.optimization.weight_decay,
+            warmup_steps=config.optimization.warmup_steps,
+            track_grad_norm=OmegaConf.select(config, "optimization.track_grad_norm", default=-1),
+            validation_metric=validation_metric,
+            validation_metric_name=self._validation_metric_name,
+            custom_metric_func=custom_metric_func,
+            loss_func=loss_func,
+            mixup_fn=mixup_func,
+            efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+            mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+            skip_final_val=OmegaConf.select(config, "optimization.skip_final_val", default=False),
+        )
+
+    def build_task_per_run(
```

Comment:
Changed it to `get_litmodule_per_run`. I use `litmodule` instead of `lit_module` to avoid the import name `lit_module` and also align with `datamodule`.